### ISSUE TITLE
Lit tests failing with STABLEHLO disabled

### ISSUE THREAD

**Issue by dbabokin** [llvm/torch-mlir#3816]
When building with `-DTORCH_MLIR_ENABLE_STABLEHLO=OFF` and running lit tests, I see three fails:
```
  TORCH_MLIR :: Conversion/TorchToLinalg/sparse.mlir
  TORCH_MLIR :: Dialect/Torch/invalid.mlir
  TORCH_MLIR :: Dialect/Torch/ops.mlir
```

All of the fails are due to `#sparse_tensor.encoding` not being recogined:
```
bin/torch-mlir-opt < ../test/Conversion/TorchToLinalg/sparse.mlir -split-input-file -verify-diagnostics
module {
}

// -----
within split at <stdin>:3 offset :3:23: error: unexpected error: #"sparse_tensor"<"encoding<{ map = (d0, d1) -> (d0 : dense, d1 : compressed) }>"> : 'none' attribute created with unregistered dialect. If this is intended, please call allowUnregisteredDialects() on the MLIRContext, or use -allow-unregistered-dialect with the MLIR opt tool used
#CSR = #sparse_tensor.encoding<{ map = (d0, d1) -> (d0 : dense, d1 : compressed) }>
```

 `#sparse_tensor.encoding` was introduced in https://github.com/llvm/torch-mlir/pull/2809 and https://github.com/llvm/torch-mlir/pull/2799 by @aartbik.

The patches have nothing specific to STABLEHLO. I'm wondering if ifdefs for STABLEHLO covering too much code somewhere else.

**Comment by aartbik** [comment]
Due to a career switch, there will be a bit of a delay, but I will have a look!

**Comment by dbabokin** [comment]
@aartbik any chance you have a look?

**Comment by aartbik** [comment]
Sorry, it is still on my todo list but give me some more time to ramp up (it it is very urgent, please let me know also)

**Comment by aartbik** [comment]
Parts of the rewriting were relying on StableHLO to register the sparse tensor dialect. Fix coming up!

**Comment by dbabokin** [comment]
Thanks, Aart!

**Comment by aartbik** [comment]
My pleasure. Apologies for the very long delay getting this done.
